apps/server/src/modules/authorization/domain/rules/context-external-tool.rule.ts

```diff
@@ -1,34 +1,34 @@
-import { Injectable } from '@nestjs/common';
 import { ContextExternalTool } from '@modules/tool/context-external-tool/domain';
 import { ContextExternalToolEntity } from '@modules/tool/context-external-tool/entity';
+import { Injectable } from '@nestjs/common';
 import { User } from '@shared/domain/entity';
-import { AuthorizationContext, Rule } from '../type';
 import { AuthorizationHelper } from '../service/authorization.helper';
+import { AuthorizationContext, Rule } from '../type';
 
 @Injectable()
-export class ContextExternalToolRule implements Rule {
+export class ContextExternalToolRule implements Rule<ContextExternalToolEntity | ContextExternalTool> {
 	constructor(private readonly authorizationHelper: AuthorizationHelper) {}
 
-	public isApplicable(user: User, entity: ContextExternalToolEntity | ContextExternalTool): boolean {
-		const isMatched: boolean = entity instanceof ContextExternalToolEntity || entity instanceof ContextExternalTool;
+	public isApplicable(user: User, object: unknown): boolean {
```

Comment:
> The Scout Principle in programming, also known as the Boy Scout Rule, is a guideline that encourages developers to always leave the code cleaner than they found it.
Phillip Wirth kindly pointed out that the object parameter being of type T does not make sense, since the point of the function is that we don't know which type is given to the function. This is only a small renaming and has to be done in every rule when the interface changes.

Reply:
The scout principle is good, unfortunately we use squashAndMerge in the project, which is committed to the wrong context